src/GridFS/GridFsUpload.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace MongoDB\GridFS;
+
+use MongoDB\Collection;
+use MongoDB\Exception\RuntimeException;
+use MongoDB\Exception\UnexpectedTypeException;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\BSON;
+/**
+ * GridFsupload abstracts the processes of inserting into a GridFSBucket
+ *
+ * @api
+ */
+class GridFsUpload extends GridFsStream
+{
+    private $ctx;
+    private $bufferLength;
+    private $indexChecker;
+    private $length=0;
+    /**
+     * Constructs a GridFS upload stream
+     *
+     * Supported options:
+     *
+     *  * contentType (string): DEPRECATED content type to be stored with the file.
+     *    This information should now be added to the metadata
+     *
+     *  * aliases (array of strings): DEPRECATED An array of aliases.
+     *    Applications wishing to store aliases should add an aliases field to the
+     *    metadata document instead.
+     *
+     *  * metadata (array or object): User data for the 'metadata' field of the files
+     *    collection document.
+     *
+     *  * writeConcern (MongoDB\Driver\WriteConcern): Write concern.
+     *
+     * @param \MongoDB\Collection    $filesCollection  Files Collection
+     * @param \MongoDB\Collection    $chunksCollection Chunks Collection
+     * @param int32                  $chunkSizeBytes   Size of chunk
+     * @param string                 $filename         Filename to insert
+     * @param array                  $options          File options
+     * @throws InvalidArgumentException
+     */
+    public function __construct(
+        Bucket $bucket,
```

Comment:
I believe this class only needs the chunk size and it was using Bucket to insert chunk and file documents. We can reduce this to just take the chunk size option, Manager instance, and GridFS prefix (e.g. "fs") directly. This will remove public methods from the Bucket and we won't need to use it as an options container.